src/Cms/App.php

```diff
@@ -184,37 +184,42 @@ public function api()
     }
 
     /**
-     * Applies a hook to the given value;
-     * the value that gets modified by the hooks
-     * is always the last argument
+     * Applies a hook to the given value
      *
      * @internal
-     * @param string $name Hook name
-     * @param mixed ...$args Arguments to pass to the hooks
+     * @param string $name Full event name
+     * @param array $args Associative array of named event arguments
+     * @param string $modify Key in $args that is modified by the hooks
+     * @param \Kirby\Cms\Event|null $originalName Event object (internal use)
      * @return mixed Resulting value as modified by the hooks
      */
-    public function apply(string $name, ...$args)
+    public function apply(string $name, array $args, string $modify, ?Event $originalEvent = null)
```

Comment:
@lukasbestle Just a quick note: We wanted to stay away from type hinting objects

Reply:
Wasn't that just about method return values though? We have quite a few methods that use class type hinting for arguments, for example in the `AppPlugins` trait. Also I think it's not a problem for `App` as that class is not going to be extended by plugins.

But if you think we should remove the type hinting, that's no problem either.

Reply:
@distantnative Ping :)
As I said, either way is fine for me, but we need to decide.